src/sql/workbench/contrib/welcome/page/browser/gettingStartedTour.ts

```diff
@@ -0,0 +1,358 @@
+
+let guidedTour: GuidedTour;
+
+export class GuidedTourAction extends Action {
+	public static readonly ID = 'workbench.action.createGuidedTour';
+	public static readonly LABEL = localize('guidedTour', "User Welcome Tour");
+
+	constructor(
+		id: string,
+		label: string,
+		@IInstantiationService private readonly instantiationService: IInstantiationService
+	) {
+		super(id, label);
+	}
+
+	public run(): Promise<void> {
+		if (!guidedTour) {
+			guidedTour = this.instantiationService.createInstance(GuidedTour);
+		}
+		guidedTour.create();
+		return Promise.resolve();
+	}
+}
+
+export class HideGuidedTourAction extends Action {
+	public static readonly ID = 'workbench.action.hideGuidedTour';
+	public static readonly LABEL = localize('hideGuidedTour', "Hide Welcome Tour");
+
+	constructor(
+		id: string,
+		label: string
+	) {
+		super(id, label);
+	}
+
+	public run(): Promise<void> {
+		if (guidedTour) {
+			guidedTour.hide();
+		}
+		return Promise.resolve();
+	}
+}
+
+export class GuidedTour extends Disposable {
+	private _overlayVisible: IContextKey<boolean>;
+	private _overlay!: HTMLElement;
+
+	constructor(
+		@ILayoutService private readonly layoutService: ILayoutService,
+		@IContextKeyService private readonly _contextKeyService: IContextKeyService,
+	) {
+		super();
+		this._overlayVisible = IS_OVERLAY_VISIBLE.bindTo(this._contextKeyService);
+	}
+
+	public create(): void {
+		const offset = this.layoutService.offset?.top ?? 0;
+		const tourElements = [];
+		this._overlay = dom.append(this.layoutService.container, $('.ads_tour'));
+		this._overlay.style.top = `${offset}px`;
+		this._overlay.style.height = `calc(100% - ${offset}px)`;
+		this._overlay.style.display = 'none';
+		this._overlay.tabIndex = -1;
+
+		this._overlay.addEventListener('keydown', (e) => {
+			let event = new StandardKeyboardEvent(e);
+			if (event.equals(KeyCode.Escape)) {
+				this.hide();
+			}
+		});
+
+		tourData.forEach(({ key, order, header, body, step, elmClass, id, btnId, btnText, docs, elementToAppendTo, arrow, popupImage }, i): void => {
+			const container = document.createElement('div');
+			let positionVertical;
+			let positionHorizontal;
+			let subjectElement;
+			if (elementToAppendTo !== 'center') {
+				subjectElement = document.querySelector(elementToAppendTo) as HTMLElement;
+				const subjectElementPosition = subjectElement.getBoundingClientRect();
+				positionHorizontal = Math.round((subjectElementPosition.left + 70));
+				positionVertical = Math.round((subjectElementPosition.top - 22));
+				subjectElement.style.pointerEvents = 'none';
+			}
+			const flexClasses = ['flex', 'column'];
+			const btnContainer = document.createElement('div');
+			const flexContainer = document.createElement('div');
+			const headerTag = document.createElement('h1');
+			const bodyTag = document.createElement('p');
+			const stepText = document.createElement('p');
+			if (docs) {
+				const docsLink = document.createElement('a');
```

Comment:
For all these - instead of creating the element and then manually appending the classes and attributes you can use DOM.$, e.g. https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/browser/ui/actionbar/actionbar.ts#L267